sdk/eventhub/event-hubs/src/eventHubConsumerClient.ts

```diff
@@ -361,124 +317,49 @@ export class EventHubConsumerClient {
     eventProcessor.start();
 
     return {
-      isRunning: () => eventProcessor.isRunning(),
+      get isRunning() {
+        return eventProcessor.isRunning();
+      },
       close: () => eventProcessor.stop()
     };
   }
 }
 
-/**
- * @ignore
- */
-export function createPartitionProcessorType(
-  partitionManager: PartitionManager,
-  options: SubscriptionEventHandlers
-): typeof PartitionProcessor {
-  class DefaultPartitionProcessor extends PartitionProcessor {
-    private _partitionCheckpointer?: SimplePartitionCheckpointer;
-
-    async processEvents(events: ReceivedEventData[]): Promise<void> {
-      await options.processEvents(
-        events,
-        this._partitionCheckpointer!
-      );
-    }
-
-    async processError(error: Error): Promise<void> {
-      if (options.processError) {
-        await options.processError(error, {
-          partitionId: this.partitionId,
-          consumerGroupName: this.consumerGroupName,
-          eventHubName: this.eventHubName,
-          fullyQualifiedNamespace: this.fullyQualifiedNamespace
-        });
-      }
-    }
-
-    async initialize() {
-      this._partitionCheckpointer = new SimplePartitionCheckpointer(partitionManager, this, this.eventHubName, this.consumerGroupName, this.partitionId, this.fullyQualifiedNamespace);
-
-      if (options.processInitialize) {
-        await options.processInitialize({
-          partitionId: this.partitionId,
-          consumerGroupName: this.consumerGroupName,
-          eventHubName: this.eventHubName,
-          fullyQualifiedNamespace: this.fullyQualifiedNamespace
-        });
-      }
-    }
-
-    async close(reason: CloseReason) {
-      if (options.processClose) {
-        await options.processClose(reason, this._partitionCheckpointer!);
-      }
-    }
-  }
-
-  return DefaultPartitionProcessor;
-}
-
 /**
  * @internal
  * @ignore
  */
-export function isPartitionManager(
-  possible: SubscriptionOptions | string | PartitionManager
-): possible is PartitionManager {
+export function isCheckpointStore(possible: CheckpointStore | any): possible is CheckpointStore {
   if (!possible) {
     return false;
   }
 
-  const partitionManager = possible as PartitionManager;
+  const checkpointStore = possible as CheckpointStore;
 
   return (
-    typeof partitionManager.claimOwnership === "function" &&
-    typeof partitionManager.listOwnership === "function" &&
-    typeof partitionManager.updateCheckpoint === "function"
+    typeof checkpointStore.claimOwnership === "function" &&
+    typeof checkpointStore.listOwnership === "function" &&
+    typeof checkpointStore.updateCheckpoint === "function"
```

Comment:
We should also check for `listCheckpoints`

Reply:
Done.